src/libs/actions/Policy.ts

```diff
@@ -2746,6 +2746,59 @@ function createPolicyCategory(policyID: string, categoryName: string) {
     API.write(WRITE_COMMANDS.CREATE_WORKSPACE_CATEGORIES, parameters, onyxData);
 }
 
+function createPolicyTag(policyID: string, tagName: string) {
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`,
+                value: {
+                    Tag: {
+                        tags: {
+                            [tagName]: {
+                                name: tagName,
+                                enabled: false,
+                            },
+                        },
+                        errors: null,
+                        pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                    },
+                },
+            },
+        ],
+        successData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`,
+                value: {
+                    Tag: {
+                        errors: null,
+                        pendingAction: null,
+                    },
+                },
+            },
+        ],
+        failureData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.POLICY_TAGS}${policyID}`,
+                value: {
+                    Tag: {
+                        errors: ErrorUtils.getMicroSecondOnyxError('workspace.tags.genericFailureMessage'),
```

Comment:
This error should be nested in the specific created tag, not at the high level tagList object, i.e. [tagListName].tags.[tagName].errors

Reply:
The types won't allow me to do that. Should I change the types?

Reply:
yes, we need to update them

Reply:
Handled.